app/controllers/api/export_solutions_controller.rb

```diff
@@ -0,0 +1,25 @@
+module API
+  class ExportSolutionsController < BaseController
+    before_action :use_track
+    before_action :use_exercise
+
+    def index
+      return render_403(:solutions_export_not_accessible) unless current_user.maintainer? || current_user.admin?
+
+      zip_file_stream = Exercise::ExportSolutionsToZipFile.(@exercise)
+
+      send_data zip_file_stream, type: 'application/zip', filename: 'solutions.zip'
+    end
+
+    private
+    def use_track
+      @track = Track.find_by(slug: params[:track_slug])
+      return render_track_not_found if @track.blank?
```

Comment:
You didn't like my suggestion of `find` and the rescue?

Reply:
I missed the rescue part. I'll update